README.md

```diff
@@ -27,9 +28,27 @@ For `scala`, `kotlin`, `kotlinscript`, `vue` & `graphql` languages, you'll need
 
 ## Usage
 
-Before using this extension, ensure you've [configured Spotless](https://github.com/diffplug/spotless/tree/master/plugin-gradle) correctly in your Gradle build file.
+Before using this extension ensure you've [configured Spotless](https://github.com/diffplug/spotless/tree/master/plugin-gradle) correctly in your Gradle build file.
 
-### Format on Save
+### Diagnostics
+
+Diagnostics are reported by generating a diff from the `spotlessApply` result.
+
+Quick fixes can be used to fix separate parts of the document.
+
+### Formatting
+
+Formatting uses diagnostic information (if available) to quick format your code. You can help ensure diagnostic information is available by disabling `files.trimTrailingWhitespace` in your settings.json:
```

Comment:
I've added a note to change the logic so we can compare sources while ignoring changes done by 3rd party code actions. When i've implemented that logic we won't need the user to adjust any settings.